test/unit/model/upsert.test.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+/* jshint -W030 */
+var chai = require('chai')
+  , Support = require(__dirname + '/../support')
+  , current = Support.sequelize
+  , sinon = require('sinon')
+  , Promise = current.Promise
+  , DataTypes = require('../../../lib/data-types');
+
+describe(Support.getTestDialectTeaser('Model'), function() {
+
+  if (current.dialect.supports.upserts) {
+    describe('method upsert', function () {
+      var User = current.define('User', {
+        name: DataTypes.STRING,
+        secretValue: {
+          type: DataTypes.INTEGER,
+          allowNull: false
+        }
+      });
+
+      before(function () {
+        this.query = current.query;
+        current.query = sinon.stub().returns(Promise.resolve());
+      });
+
+      after(function () {
+        current.query = this.query;
+      });
+
+      it('skip validations for missing fields', function() {
+        return User.upsert({
+          name: 'Grumpy Cat'
+        });
```

Comment:
True, its just not immediately obvious to see - maybe `expect(..).not.to.be.rejectedWith(current.ValidationError)`